### DISCUSSION TITLE
What about support for working with LiteStar?

### DISCUSSION THREAD

**Discussion by pikhovkin** [ag2ai/faststream#769] [Ideas]
What about support for working with [LiteStar](https://litestar.dev/)? [Benchmarks](https://docs.litestar.dev/latest/benchmarks.html#json)

**Comment by sternakt** [comment]
To support Litestar, the approach aligns closely with the logic used for FastAPI. In order to utilize your FastStream brokers, it is necessary to provide a context manager responsible for managing your broker's startup and shutdown as an argument to the Litestar lifespan parameter. You can find more details about this in the [Litestar documentation on lifespan context managers](https://docs.litestar.dev/latest/usage/applications.html#lifespan-context-managers). Following this setup, you can execute your Litestar application as usual, and the broker will seamlessly start alongside it.

Here is and example on how it could be done:

```python
from contextlib import asynccontextmanager

from litestar import Litestar, get

from faststream.kafka import KafkaBroker

# Define your broker and async endpoints
broker = KafkaBroker("localhost:9092")

@broker.subscriber("test")
async def base_handler(body):
    print(body)

# Create a context manager for broker startup/shutdown
@asynccontextmanager
async def lifespan(app: Litestar):
    await broker.start()
    yield
    await broker.close()

# Define your Litestar application and pass the broker context manager to its lifespan
@get("/")
def read_root():
    return {"Hello": "World"}
    
app = Litestar([get], lifespan=[lifespan])
```

**Reply by Lancetnik** [reply]
And an another one example using `on_startup` and `on_shutdown` hooks:

```python
from litestar import Litestar, get
from faststream.rabbit import RabbitBroker

broker = RabbitBroker("amqp://guest:guest@localhost:5672/")

@broker.subscriber("queue")
async def handle(msg):
    print(msg)

@get("/")
async def index() -> str:
    return "Hello, world!"

app = Litestar(
    [index],
    on_startup=(broker.start,),
    on_shutdown=(broker.close,),
)

```
